.github/actions/generate-coverage/tests/test_scripts.py

```
@@ -0,0 +1,87 @@
+    assert "Coverage" in res.stdout
+
+    calls = shell_stubs.calls_of("cargo")
+    assert len(calls) == 1
+    assert "--output-path" in calls[0].argv
+
+    data = gh.read_text().splitlines()
+    assert f"file={out}" in data
+    assert "percent=81.5" in data
+
+
+def test_run_rust_failure(tmp_path, shell_stubs):
```

Ruff (0.11.9)

Missing return type annotation for public function `test_run_rust_failure`

Add return type annotation: `None`

(ANN201)